src/EventStore.Core/TransactionLog/Chunks/TFChunk/TFChunk.cs

```diff
@@ -381,30 +378,9 @@ private enum CacheStatus {
 		}
 
 		private ReaderWorkItem CreateInternalReaderWorkItem() {
-			Stream stream;
-			if (_unbuffered) {
-				stream = UnbufferedFileStream.Create(
-					_filename,
-					FileMode.Open,
-					FileAccess.Read,
-					FileShare.ReadWrite,
-					false,
-					1024 * 1024,
-					4096,
-					false,
-					4096);
-			} else {
-				stream = new FileStream(
-					_filename,
-					FileMode.Open,
-					FileAccess.Read,
-					FileShare.ReadWrite,
-					ReadBufferSize,
-					_reduceFileCachePressure ? FileOptions.None : FileOptions.RandomAccess);
-			}
+			Debug.Assert(_handle is not null);
 
-			var reader = new BinaryReader(stream);
-			return new ReaderWorkItem(stream, reader, false);
+			return new ReaderWorkItem(_handle, _unbuffered);
```

Comment:
This change seems to alter the semantics of the `--unbuffered` configuration option (defined [here](https://github.com/EventStore/EventStore/blob/fb20c2cb507f4ceaf5216d4a43736967de032a80/src/EventStore.Core/ClusterVNodeOptions.cs#L462)) as we no longer bypass the file caches when `unbuffered` is true here. We may consider changing the definition or removing the configuration option and document it as a breaking change.

Reply:
`--unbuffered` and `--writeThrough` semantics still preserved. If `_unbuffered=false`, `BufferedStream` is created to wrap unbuffered stream, otherwise not.

Reply:
`--writeThrough` now is working exactly as expected. In the previous version, `FileStream` created for the writer side didn't take into account this flag using `FileOptions.WriteThrough` flag. Now this option is considered when file handle needs to be created.

Reply:
> `--unbuffered` and `--writeThrough` semantics still preserved. If `_unbuffered=false`, `BufferedStream` is created to wrap unbuffered stream, otherwise not.

When `--unbuffered` is specified, does it still use the `O_DIRECT` flag on linux / `FILE_FLAG_NO_BUFFERING` on windows?

i believe this is what is intended when `Direct I/O` is mentioned in the configuration option description:
`Enables Unbuffered/DirectIO when writing to the file system, this bypasses filesystem caches.`

Reply:
> does it still use the O_DIRECT flag on linux 

It's not. `O_SYNC` is actually used when `WriteThrough` applied.

> O_DIRECT alone only promises that the kernel will avoid copying data from user space to kernel space, and will instead write it directly via DMA (Direct memory access; if possible). Data does not go into caches. There is no strict guarantee that the function will return only after all data has been transferred.

In the same time

> O_SYNC guarantees that the call will not return before all data has been transferred to the disk (as far as the OS can tell)

Both still provide weak guarantees. IMHO, `O_SYNC` has stronger guarantees rather than `O_DIRECT`.